ksql-common/src/main/java/io/confluent/ksql/function/TableFunctionFactory.java

```diff
@@ -55,4 +56,24 @@ public String getVersion() {
   public boolean isInternal() {
     return metadata.isInternal();
   }
+
+  public synchronized void eachFunction(final Consumer<KsqlTableFunction> consumer) {
```

Comment:
is there any reason why this is synchronized? nothing else in the class seems to require thread safety

Reply:
I believe this is because udfIndex is not threadsafe and TableFunctionFactory can be accessed by different threads. There are other methods that need to be synchronized too, I'll change them.